public/util/lex.ts

```diff
@@ -33,129 +75,288 @@ export function variablesToLexLanguage(variables: Variable[]): Lex {
+        highlight.value.maxX = parseFloat(lq.maxX.key);
+        highlight.value.minY = parseFloat(lq.minY.key);
+        highlight.value.maxY = parseFloat(lq.maxY.key);
+      } else if (type === DATETIME_FILTER) {
+        highlight.value.min = dateToNum(lq.min);
+        highlight.value.max = dateToNum(lq.max);
+        highlight.type = DATETIME_FILTER;
+      } else if (isNumericType(type)) {
+        highlight.value.min = parseFloat(lq.min.key);
+        highlight.value.max = parseFloat(lq.max.key);
+        highlight.type = NUMERICAL_FILTER;
+      } else {
+        highlight.value = lq.value.key;
+      }
+    }
   });
-  return filters;
+  return {
+    filters: filters,
+    highlight: highlight,
+  };
+}
+
+function modeToRelation(mode: string): ValueStateValue {
+  switch (mode) {
+    case INCLUDE_FILTER:
+      return distilRelationOptions[0];
+    case EXCLUDE_FILTER:
+      return distilRelationOptions[1];
+    case HIGHLIGHT:
+      return distilRelationOptions[2];
+    default:
+      return distilRelationOptions[1];
+  }
 }
 
 function variablesToLexSuggestions(variables: Variable[]): ValueStateValue[] {
   if (!variables) return;
-  return variables.map((variable) => {
-    const name = variable.colDisplayName;
+
+  return variables.reduce((a, v) => {
+    const name = v.key;
     const options = {
-      type: colTypeToOptionType(variable.colType.toLowerCase()),
+      type: colTypeToOptionType(v.colType.toLowerCase()),
     };
-    return new ValueStateValue(name, options);
-  });
+    const config = {
+      displayKey: v.colDisplayName,
+    };
+    a.push(new ValueStateValue(name, options, config));
+
+    if (v.distilRole === "grouping") {
+      switch (v.colType) {
+        case TIMESERIES_TYPE:
+          const grouping = v.grouping as TimeseriesGrouping;
+          a.push(
+            new ValueStateValue(
+              grouping.xCol,
+              { type: DATETIME_FILTER },
+              { displayKey: grouping.xCol }
+            )
+          );
+          break;
+        case MULTIBAND_IMAGE_TYPE:
+        case GEOBOUNDS_TYPE:
+        case GEOCOORDINATE_TYPE:
+          break;
+        default:
+          console.log("unknown grouped type");
+      }
+    }
+
+    return a;
+  }, []);
 }
 
 function colTypeToOptionType(colType: string): string {
-  if (isNumericType(colType)) {
-    return NUMERICAL_FILTER;
+  if (colType.toLowerCase() === GEOBOUNDS_TYPE) {
+    return GEOBOUNDS_FILTER;
   } else if (colType.toLowerCase() === DATE_TIME_LOWER_TYPE) {
     return DATETIME_FILTER;
-  } else if (colType === CATEGORICAL_TYPE) {
+  } else if (isNumericType(colType)) {
+    return NUMERICAL_FILTER;
+  } else if (colType === CATEGORICAL_TYPE || colType === TIMESERIES_TYPE) {
     return CATEGORICAL_FILTER;
   } else if (isTextType(colType)) {
     return TEXT_FILTER;
   } else {
     return TEXT_FILTER;
   }
 }
+
+function buildVariableDictionary(variables: Variable[]) {
```

Comment:
Ok.

Reply:
That's not a lex dictionary though, that's just a dictionary of distil variables for current dataset I built so that we can do faster look ups for turning filter parameters in to a lex bar query.

Reply:
That's what I thought, having this comment will make things easier to debug later if necessary.